src/main/java/org/broadinstitute/consent/http/service/InstitutionService.java

```diff
@@ -86,6 +93,27 @@ public Institution findInstitutionById(Integer id) {
     return institution;
   }
 
+  private InstitutionDomainMap getInstitutionDomainMap() {
+    try {
+      return store.readJsonFileFromBucket("institution-domain/allowlist.json",
+          InstitutionDomainMap.class);
+    } catch (IOException e) {
+      throw new ServerErrorException("Could not load institution configuration",
+          HttpStatusCodes.STATUS_CODE_SERVER_ERROR, e);
+    }
+  }
+
+  public Institution findInstitutionForEmail(String email) {
+    String name = getInstitutionDomainMap().getInstitutionForEmail(email);
+    if (name != null) {
+      var institutions = institutionDAO.findInstitutionsByName(name);
+      if (institutions.size() == 1) {
+        return institutions.get(0);
```

Comment:
If we have multiple matches, should we throw an error instead? We've agreed that this case should never happen, so I think it would make sense to encode in our system.

Reply:
I agree that it should be checked, but if it's checked here, that means we won't know about the problem until a user whose domain has this issue tries to create an account.

If we wanted to catch it earlier, we'd want to check for duplicates whenever the JSON data changes. Maybe we should have an upload/deploy process that checks the data before copying to the bucket?